packages/hardhat-core/test/internal/core/secrets/secrets-manager.ts

```diff
@@ -0,0 +1,126 @@
+import fs from "fs-extra";
+import { expect } from "chai";
+import { SecretsManager } from "../../../../src/internal/core/secrets/secrets-manager";
+
+const TMP_FILE_PATH = `${__dirname}/test-secrets.json`;
```

Comment:
This is better done with [`os.tmpdir`](https://nodejs.org/api/os.html#ostmpdir) in a `before` or `beforeEach`.